src/components/ProgramRecordSendModal/SendLearnerRecordModal.jsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ModalDialog, Form, SelectableBox, ActionRow, Button,
+} from '@edx/paragon';
+
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { logError } from '@edx/frontend-platform/logging';
+
+import sendRecords from './data/service';
+
+const SendLearnerRecordModal = ({
+  isOpen, toggleSendRecordModal, creditPathways, uuid, username, setSendRecord,
+}) => {
+  const [selectedPathways, setSelectedPathways] = useState([]);
+
+  const handleSendRecords = async () => {
+    const pathwaysToSendRecords = creditPathways.filter(pathway => selectedPathways.includes(pathway.name));
+
+    const orgs = {
+      sendRecordSuccessOrgs: [],
+      sendRecordFailureOrgs: [],
+    };
+
+    await Promise.all(pathwaysToSendRecords.map(pathway => sendRecords(uuid, username, pathway.id)
+      .then(response => {
+        if (response.status >= 200 && response.status < 300) {
+          orgs.sendRecordSuccessOrgs.push(pathway);
+        } else {
+          orgs.sendRecordFailureOrgs.push(pathway);
+        }
+      })
+      .catch(error => {
+        orgs.sendRecordFailureOrgs.push(pathway);
+        const errorMessage = (`Error: Could not send ${pathway.name} record: ${error.message}`);
+        logError(errorMessage);
+      })));
+
+    setSendRecord(prev => ({
+      ...prev,
+      ...orgs,
+    }));
+
+    toggleSendRecordModal();
+  };
+
+  const handleCheckboxChange = (e) => {
+    e.persist();
+
+    if (e.target.checked) {
+      setSelectedPathways(prev => ([
+        ...prev,
+        e.target.value,
+      ]));
+    } else {
+      setSelectedPathways(prev => ([
+        ...prev.filter(pathway => pathway !== e.target.value),
+      ]));
+    }
+  };
+
+  return (
+    <>
+      <ModalDialog
+        title="Send Program Record"
+        isOpen={isOpen}
+        hasCloseButton
+        onClose={toggleSendRecordModal}
+        size="lg"
+        className="send-record-modal"
+        isFullscreenScroll
+      >
+        <ModalDialog.Header>
+          <ModalDialog.Title>
+            <FormattedMessage
+              id="send.record.modal.header"
+              defaultMessage="Send Program Record to edX Credit Partner"
```

Comment:
Hm. I think we have a problem here. We have "edX" hardcoded into the formatted message. This is potentially going to be used by Open edX and the message being "edX credit partner" is likely not what we want. We need the ability for this message to be a bit more flexible.

In the original component, we pulled in the `platform` name so that it matches the (Open edX) instance's site configuration/name.

Reply:
See the original component: https://github.com/openedx/credentials/blob/dc5f397fbc67ad2b2304b1dd91a9879f35f22bb2/credentials/static/components/SendLearnerRecordModal.jsx#L92

Reply:
Thanks for catching these! I definitely overlooked that but luckily the `FormattedMessage` component provides us with a pretty easy fix